pkg/provision/providers/qemu/node.go

```diff
@@ -368,3 +365,30 @@ func (p *provisioner) createMetalConfigISO(state *vm.State, nodeName, config str
 
 	return isoPath, nil
 }
+
+func getLaunchNetworkConfigBase(state *vm.State, clusterReq provision.ClusterRequest, nodeReq provision.NodeRequest) networkConfigBase {
+	return networkConfigBase{
+		BridgeName:   state.BridgeName,
+		CIDRs:        clusterReq.Network.CIDRs,
+		IPs:          nodeReq.IPs,
+		GatewayAddrs: clusterReq.Network.GatewayAddrs,
+		MTU:          clusterReq.Network.MTU,
+		Nameservers:  clusterReq.Network.Nameservers,
+	}
+}
+
+// getRandomMacAddress generates a random local MAC address
+// https://stackoverflow.com/a/21027407/10938317
+func getRandomMacAddress() string {
+	const (
+		local     = 0b10
+		multicast = 0b1
+	)
+
+	buf := make([]byte, 6)
+	rand.Read(buf) //nolint:errcheck
+	// clear multicast bit (&^), ensure local bit (|)
+	buf[0] = buf[0]&^multicast | local
+
+	return fmt.Sprintf("%02x:%02x:%02x:%02x:%02x:%02x", buf[0], buf[1], buf[2], buf[3], buf[4], buf[5])
```

Comment:
https://pkg.go.dev/net#HardwareAddr

```suggestion
	return net.HardwareAddr(buf).String()
```